### ISSUE TITLE
Ansible version check in provisioner does not work since Ansible 2.10

### ISSUE THREAD

**Issue by haxorof** [hashicorp/vagrant#12204]
### Vagrant version

```
Vagrant 2.2.14
```

### Host operating system

```
Windows 10 20H2 (OS Build 19042.804)
```

### Guest operating system

Ubuntu 20.04.2 LTS (Focal Fossa)

### Vagrantfile

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure("2") do |config|
  config.vm.box = "generic/ubuntu2004"
  config.vagrant.plugins = ["vagrant-vbguest"]
  config.vm.synced_folder ".", "/vagrant", :mount_options => ["rw"]

  config.vm.provision "ansible_local" do |ansible|
    ansible.install_mode = "pip"    
    ansible.playbook = "playbook.yml"
    ansible.version = "2.10.7"
    ansible.pip_install_cmd = "sudo apt-get install -y python3-pip python-is-python3 haveged && sudo ln -s -f /usr/bin/pip3 /usr/bin/pip"
  end
end
```

Ansible playbook in the same directory:
```yaml
---
- hosts: all
  tasks:
    - debug:
        msg: test
```

### Expected behavior

Ansible 2.10.7 installed and playbook run.

### Actual behavior

Ansible version check fails because since Ansible 2.10 there has been a split of the code base so now `ansible-base` package is what version that is shown and `ansible` package depends on `ansible-base` plus community collection.

```
==> default: Running provisioner: ansible_local...
    default: Installing Ansible...
    default: Installing pip... (for Ansible installation)
The requested Ansible version (2.10.7) was not found on the guest.
Please check the Ansible installation on your Vagrant guest system (currently: 2.10.6),
or adapt the provisioner `version` option in your Vagrantfile.
See https://docs.vagrantup.com/v2/provisioning/ansible_common.html#version
for more information.
```

Ansible version command show `ansible-base` version:
```shell
vagrant@ubuntu2004:~$ ansible --version
ansible 2.10.6
  config file = None
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]
```

What actually happened?

### Steps to reproduce

1. vagrant up
2. see that it fails with the error described above

### References

- https://pypi.org/project/ansible/
- https://pypi.org/project/ansible-base/

**Comment by jotbe** [comment]
TL;DR:
I ran into the same issue, found the corresponding lines in the source code and came up with a workaround which allows installing a specific Ansible version, at a cost that the version check is basically ignored - here, applied to your example:

```yml
Vagrant.configure("2") do |config|
  config.vm.box = "generic/ubuntu2004"
  config.vagrant.plugins = ["vagrant-vbguest"]
  config.vm.synced_folder ".", "/vagrant", :mount_options => ["rw"]

  config.vm.provision "ansible_local" do |ansible|
    ansible.install_mode = "pip_args_only"
    ansible.playbook = "playbook.yml"
    #ansible.version = "2.10.7"
    ansible.pip_install_cmd = "sudo apt-get install -y python3-pip python-is-python3 haveged && sudo ln -s -f /usr/bin/pip3 /usr/bin/pip"
    ansible.pip_args = "ansible==2.10.7"
  end
end
```

**Details:**

- Version is gathered by `ansible --version`: https://github.com/hashicorp/vagrant/blob/5b501a3fb05ed0ab16cf10991b3df9d231edb5cf/plugins/provisioners/ansible/provisioner/guest.rb#L74
- called from: https://github.com/hashicorp/vagrant/blob/5b501a3fb05ed0ab16cf10991b3df9d231edb5cf/plugins/provisioners/ansible/provisioner/base.rb#L55
- checked/compared here: https://github.com/hashicorp/vagrant/blob/5b501a3fb05ed0ab16cf10991b3df9d231edb5cf/plugins/provisioners/ansible/provisioner/guest.rb#L64

I used that [workaround for one of my VMs to install Ansible 3.3.0.](https://git.jotbe.io/jotbe/ansible-devops-vm/commit/972e6c0d87d09cca4397fc8eba0885de04046673)

**Comment by T2L** [comment]
I can confirm the issue. Workaround from @jotbe works pretty good. Thanks!

**Comment by soapy1** [comment]
This issue should be fixed by https://github.com/hashicorp/vagrant/pull/13054. It will be included in the next release of Vagrant 🙌
